sdk/metricsadvisor/Azure.AI.MetricsAdvisor/README.md

```diff
@@ -159,7 +159,7 @@ string sqlServerConnectionString = "<connectionString>";
 string sqlServerQuery = "<query>";
 
 var dataFeedName = "Sample data feed";
-var dataFeedSource = new MySqlDataFeedSource(sqlServerConnectionString, sqlServerQuery);
+var dataFeedSource = new SqlServerDataFeedSource(sqlServerConnectionString, sqlServerQuery);
```

Comment:
I am curious on how this was working? was the data also available in mysql? or how does the datasource affects the datafeed?

Reply:
Both [MySqlDataFeedSource](https://github.com/Azure/azure-sdk-for-net/blob/d15443a83fbfbf0032295d5829a53b1aa30d18c2/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedSources/MySqlDataFeedSource.cs#L27) and [SqlServerDataFeedSource](https://github.com/Azure/azure-sdk-for-net/blob/d15443a83fbfbf0032295d5829a53b1aa30d18c2/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedSources/SqlServerDataFeedSource.cs#L27) use a `SqlSourceParameter` internally, so they require the exact same parameters (a connection string and a query). This, plus the fact that I'm not familiar with SQL stuff, was enough to introduce this bug.

We were sending a `MySqlDataFeed` JSON object to the service. It was in the expected format, but it had an invalid connection string (connection string to a SQL server). This doesn't prevent the service from creating the data feed. The sample worked just fine because the data feed was being created and an ID was being returned. You can even create detection and alert configurations. However, at some point you'll notice that anomalies are not being detected. In fact, data is not being ingested at all.

When listing ingestion statuses, we get:
```
Timestamp: 1/3/2020 12:00:00 AM +00:00
Status: Failed
Service message: Cannot fetch data from source.
```

One may argue that we should fail to create a data feed if data source credentials are not valid. However, there are two reasons for not doing so:
1. Credentials can be updated, so you can rely on `UpdateDataFeed` to fix an incorrectly inserted connection string. This wouldn't help us in this scenario because you can't change the type of data source (MySQL => SQL server) after data feed creation.
2. You can schedule data ingestion to start at some point in the future, so you won't know until then whether your credentials are valid or not.

Reply:
Agree. And in this case it should be the service validating the authentication and not us, so we are doing everything we can.
Thanks for the detail explanation